client/src/modules/events/pages/eventPage.tsx

```diff
@@ -102,15 +85,41 @@ export const EventPage: NextPage = () => {
     }
   };
 
-  const rsvps = data.event.rsvps.filter((r) => !r.on_waitlist);
-  const waitlist = data.event.rsvps.filter((r) => r.on_waitlist);
   const checkOnRsvp = async (add: boolean) => {
     if (!user) {
       return handleLoginUserFirst();
     }
 
     await onRsvp(add);
   };
+  const userRsvped = useMemo(() => {
+    const rsvp = data?.event?.rsvps.find((rsvp) => rsvp.user.id === user?.id);
+    if (!rsvp) return null;
+    return rsvp.on_waitlist ? 'waitlist' : 'rsvp';
+  }, [data?.event]);
+
+  useEffect(() => {
+    if (router.query?.emaillink && user && !userRsvped) {
+      checkOnRsvp(true);
+      router.replace('/events/' + id, undefined, { shallow: true });
+    }
+  }, [userRsvped, router.query, user]);
```

Comment:
```suggestion
  const allDataLoaded = !loading && !!user;
  const canCheckRsvp = router.query?.emaillink && !userRsvped;

  useEffect(() => {
    if (allDataLoaded) {
      if (canCheckRsvp) {
        checkOnRsvp(true);
      }
      router.replace('/events/' + id, undefined, { shallow: true })
    }
  }, [allDataLoaded, canCheckRsvp]);
```
I think we need to go this way.  If we don't separate `checkOnRsvp` from `router.replace`, then if a user goes to http://localhost:3000/events/id?emaillink=true *after* confirming, then the query parameter will stay.

This means that clicking cancel will automatically trigger `checkOnRsvp` and ask if they want to confirm again.

Reply:
@ojeytonwilliams , I've updated the code to handle this. Could you please review onc?